ApiClient/Source/Pages/Queries/HomePageQuery.swift

```diff
@@ -26,26 +26,41 @@
 import Foundation
 import Alamofire
 
-final class HomePageQuery: Query {
-    typealias Result = JSONPage?
-    
-    let resource: String
+final class HomePageQuery: Query {    
+    typealias Result = JSONPage
+
+    let url: URL
+    let host: String = "https://api.schedjoules.com"
     let method: HTTPMethod = .get
     let parameters: Parameters = [:]
     let headers: HTTPHeaders = ["Accept" : "application/json", "Content-Type" : "application/json"]
 
-    required init(resource: String) {
-        self.resource = resource
+    private init?(queryItems: [URLQueryItem]) {
+        // Build url from components
+        guard var urlComponents = URLComponents(string: host) else {
+            return nil
+        }
+        // Add path to the url
+        urlComponents.path = "/pages"
+        // Add query items to the url
+        urlComponents.queryItems = queryItems
+        // If the url could not be constructed, return nil
+        if urlComponents.url == nil {
+            return nil
+        }
+        self.url = urlComponents.url!
     }
     
     /// Automatically add locale and location parameter to the pages URL
-    convenience init() {
-        self.init(locale: Locale.preferredLanguages[0].components(separatedBy: "-")[0], location: Locale.current.regionCode!)
+    convenience init?() {
+        let localeQuery = URLQueryItem(name: "locale", value: Locale.preferredLanguages[0].components(separatedBy: "-")[0])
+        let locationQuery = URLQueryItem(name: "location", value: Locale.current.regionCode!)
+        self.init(queryItems: [localeQuery,locationQuery])
     }
     
     /// Manualy specify locale and location parameters
-    convenience init(locale: String, location: String){
-        self.init(resource: "pages/?locale=\(locale)?location=\(location)")
+    convenience init?(locale: String, location: String) {
+        self.init(queryItems: [URLQueryItem(name: "locale", value: locale), URLQueryItem(name: "location", value: location)])
```

Comment:
can you reformat this to put the two `URLQueryItem`s onto separate lines? This would help the reader.